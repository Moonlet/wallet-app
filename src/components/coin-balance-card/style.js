import { StyleSheet } from 'react-native';
import COLORS from '../../styles/colors';

export default StyleSheet.create({
    container: {
        padding: 0,
        paddingTop: 10,
        paddingBottom: 10,
        alignItems: 'center',
        color: '#FFF'
        //width: 200
    },
    darkerText: {
        color: COLORS.lightGray
    },
    mainText: {
        fontSize: 34
    }
});
