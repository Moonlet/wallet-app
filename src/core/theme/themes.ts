export enum Theme {
    dark = 'dark',
    light = 'light'
}
