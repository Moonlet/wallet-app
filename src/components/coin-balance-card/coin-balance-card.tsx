import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Text } from '../../library';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import BigNumber from 'bignumber.js';
import { Blockchain } from '../../core/blockchain/types';
import { Amount } from '../amount/amount';
import { Icon } from '../icon';
import { IAccountState } from '../../redux/wallets/state';
import { formatAddress } from '../../core/utils/format-address';

export interface IProps {
    onPress?: () => void;
    blockchain: Blockchain;
    selectedAccount: IAccountState;
    balance: BigNumber;
    width?: number;
    active: boolean;
    styles: ReturnType<typeof stylesProvider>;
}

export const CoinBalanceCardComponent = (props: IProps) => (
    <TouchableOpacity
        onPress={() => props.onPress()}
        style={[props.styles.container, { width: props.width }]}
    >
        <View style={props.styles.rowContainer}>
            <Text style={props.styles.account}>
                {props.selectedAccount?.name || `Account ${props.selectedAccount.index + 1}`}
            </Text>
            <Text style={props.styles.address}>{formatAddress(props.selectedAccount.address)}</Text>
        </View>
        <View style={props.styles.rowContainer}>
            <Amount
                style={[props.styles.mainText, !props.active && props.styles.darkerText]}
                amount={props.balance}
                blockchain={props.blockchain}
            />
            <Icon name="arrow-down-1" size={18} style={props.styles.icon} />
        </View>
        <Amount
            style={[props.styles.secondaryText, !props.active && props.styles.darkerText]}
            amount={props.balance}
            blockchain={props.blockchain}
            convert
        />
    </TouchableOpacity>
);

export const CoinBalanceCard = withTheme(stylesProvider)(CoinBalanceCardComponent);
