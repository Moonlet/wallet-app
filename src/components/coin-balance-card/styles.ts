import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            alignSelf: 'center',
            alignItems: 'center'
        },
        darkerText: {
            color: theme.colors.textSecondary
        },
        rowContainer: {
            flexDirection: 'row',
            marginBottom: BASE_DIMENSION
        },
        account: {
            fontSize: 15,
            lineHeight: 20,
            color: theme.colors.text,
            marginRight: BASE_DIMENSION
        },
        address: {
            fontSize: 15,
            lineHeight: 20,
            color: theme.colors.accent
        },
        mainText: {
            fontSize: 30,
            lineHeight: 41,
            fontWeight: 'bold',
            letterSpacing: 0.4,
            color: theme.colors.text,
            marginRight: BASE_DIMENSION * 2
        },
        secondaryText: {
            fontSize: 16,
            lineHeight: 21,
            color: theme.colors.textSecondary
        },
        icon: {
            alignSelf: 'center',
            color: theme.colors.accent,
            fontWeight: 'bold'
        }
    });
