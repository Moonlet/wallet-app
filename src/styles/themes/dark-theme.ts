import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
