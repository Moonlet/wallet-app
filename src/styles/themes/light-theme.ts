import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
