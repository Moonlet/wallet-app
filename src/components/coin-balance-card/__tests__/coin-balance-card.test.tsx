import 'react-native';
import React from 'react';
import { CoinBalanceCardComponent, IProps } from '../coin-balance-card';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';
import BigNumber from 'bignumber.js';

const props: IProps = {
    currency: 'ZIL',
    balance: new BigNumber(100),
    toCurrency: 'ETH',
    width: 100,
    active: true,
    styles: styleProvider(darkTheme)
};

describe('coin balance component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<CoinBalanceCardComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
