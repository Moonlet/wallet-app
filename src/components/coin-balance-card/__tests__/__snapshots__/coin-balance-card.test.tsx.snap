// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coin balance component renders correctly 1`] = `
"<TouchableOpacity onPress={[Function: onPress]} style={{...}} activeOpacity={0.2}>
  <View style={{...}}>
    <withTheme(TextComponent) style={{...}}>
      Account 2
    </withTheme(TextComponent)>
    <withTheme(TextComponent) style={{...}}>
      zil1v...3k21b
    </withTheme(TextComponent)>
  </View>
  <View style={{...}}>
    <Connect(AmountComponent) style={{...}} amount=\\"100\\" token=\\"ZIL\\" tokenDecimals={12} blockchain=\\"ZILLIQA\\" />
    <Icon name=\\"chevron-down\\" size={18} style={{...}} allowFontScaling={false} />
  </View>
  <Connect(AmountComponent) style={{...}} amount=\\"100\\" token=\\"ZIL\\" tokenDecimals={12} blockchain=\\"ZILLIQA\\" convert={true} />
</TouchableOpacity>"
`;
